Tests/Unit/MSFT_xFirewall.Tests.ps1

```diff
@@ -149,6 +149,7 @@ try
                 }
             }
             Context 'Ensure is Present and the Firewall is Absent' {
+                Mock Get-FirewallRule
```

Comment:
Ok, I've figured it out.

Without the Mock the Test-FirewallProperty would fire because the rule existed. Although in this test we're trying to ensure the function returns $true (there is a difference). But some rules will report no difference if the rule exists because all values in the rule are just the default. So whether this test passed or not (without the new mock) would be down to the rule tested. Introducing the Mock corrected the problem.

What I did to figure this out was wrap the entire unit test code block in a foreach that tested every single rule on my computer (we could do this for real if we wanted, but it will take hours to run- if not days on AppVeyor). Without your the new Mock some rules would fail which allowed me to track and identify it what was going on.

So you definitely identified and fixed a nasty bug in the Unit tests there. The actual logic in the resource is OK.